includes/embeds/class-amp-pinterest-embed-handler.php

```diff
@@ -12,7 +12,7 @@
  */
 class AMP_Pinterest_Embed_Handler extends AMP_Base_Embed_Handler {
 
-	const URL_PATTERN = '#https?://(www\.)?pinterest\.com/pin/.*#i';
+	const URL_PATTERN = '#https?://(?:www\.)?(?:[a-z]{2}\.)?pinterest\.[a-z.]+/pin/[^/]+/?#i';
```

Comment:
Note also that the `data-url` attribute has [no validation constraints](https://github.com/ampproject/amphtml/blob/ad3ccbedcc83e98fa1c30ada12ca5e6bb9869204/extensions/amp-pinterest/validator-amp-pinterest.protoascii#L37-L41) enforced on it by the validator spec. So passing through arbitrary URLs to `<amp-pinterest>` won't cause validation errors.

We could potentially modify the regex to restrict the segment to be digits. This would make sense if we know for sure that a `pinId` is always only digits. But according to the [Pinterest docs](https://developers.pinterest.com/docs/api/pins/):

![image](https://user-images.githubusercontent.com/134745/91916586-55e94800-ec72-11ea-9100-49b308428e96.png)

Apparently it could be letters as well. So if we wanted to be more restrictive:

```suggestion
	const URL_PATTERN = '#https?://(?:www\.)?(?:[a-z]{2}\.)?pinterest\.[a-z.]+/pin/[a-z0-9]+/?#i';
```

I don't see this as being necessarily more secure. It could just prevent bogus pin IDs from being passed to the API at `https://widgets.pinterest.com/v3/pidgets/pins/info/`, which I sure hope as its own input sanitization in place!